packages/studio-plugin/src/createStudioStylingPlugin.ts

```diff
@@ -0,0 +1,61 @@
+import { PluginOption } from "vite";
+import upath from "upath";
+import { fileURLToPath } from "url";
+const currentDir = upath.dirname(fileURLToPath(import.meta.url));
+const packagesDir = upath.resolve(currentDir, "../..");
+
+/**
+ * Adds the ?inline query parameter to all CSS imports
+ * unless they are CSS from Studio, which is determined
+ * from the path of its importer. This prevents user styling
+ * from affecting Studio UI.
+ */
+export default function createStudioStylingPlugin(): PluginOption {
+  return {
+    name: "yext-studio-style-plugin",
+    enforce: "pre",
+    async resolveId(id, importer) {
+      if (!importer) {
+        return;
+      }
+      if (isImportedByStudio(importer)) {
+        return;
+      }
+      if (isStyleFile(id)) {
+        const resolvedId = (
+          await this.resolve(id, importer, { skipSelf: true })
+        )?.id;
+        if (resolvedId) {
+          return addQueryParameter(resolvedId, "inline");
+        }
+        throw new Error(
+          "Styling file could not be resolved by yext-studio-style-plugin"
+        );
+      }
+    },
+  };
+}
+
+function isImportedByStudio(importer: string) {
+  const unixImporter = upath.toUnix(importer);
+  return !!(
+    unixImporter.match(/.*\/node_modules\/@yext\/studio.*/) ||
+    unixImporter.startsWith(packagesDir)
```

Comment:
nit: if you switch the order of these two checks, you should be able to remove the double negation

Reply:
Out of curiosity, what is changed by the difference in ordering?  My understanding was that if the first is true, return the first, else return the second.

Reply:
I think for cleanliness, it would be nice to at least negate the `match` statement so we always return a boolean.

Reply:
ah nvm I agree